django_netjsonconfig/base/template.py

```diff
@@ -1,15 +1,26 @@
+from collections import OrderedDict
+
 from django.core.exceptions import ValidationError
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
+from jsonfield import JSONField
 from taggit.managers import TaggableManager
 
 from ..settings import DEFAULT_AUTO_CERT
+from ..utils import get_random_key
+from ..validators import key_validator
 from .base import BaseConfig
 
 TYPE_CHOICES = (
     ('generic', _('Generic')),
     ('vpn', _('VPN-client')),
 )
+FLAG_CHOICES = (
+    ('private', _('Private')),
+    ('public', _('Public')),
+    ('shared_secret', _('Shared Secret')),
+    ('import', _('Import'))
```

Comment:
Ok i will include that here. Was thinking of adding this to the controller instead. meaning i have to edit [issue #95](https://github.com/openwisp/openwisp-controller/issues/95).
Please i which to know if this will mean we will have two URL fields? where;
-one is used to url of the imported template
-the other used to store the url of the public and secret share templates.

Reply:
don't we only need the URL when we need to import?
When the system shares the template, the system itself (the API) generates the URL.
In that case the generated URL can be a readonly field in the admin, not a concrete database field.